File: ios/Classes/Domain/UseCase/GetResponseUseCase.swift
import Foundation

/// Use case for getting task response
struct GetResponseUseCase {
    private let repository: TaskRepository
    
    init(repository: TaskRepository) {
        self.repository = repository
    }
    
    func execute(requestId: String) async throws -> TaskInfo? {
        return try await repository.getTaskResponse(requestId: requestId)
    }
}

File: android/src/main/kotlin/com/tsitser/background_http_plugin/domain/usecase/GetResponseUseCase.kt
package com.tsitser.background_http_plugin.domain.usecase

import com.tsitser.background_http_plugin.domain.entity.TaskInfo
import com.tsitser.background_http_plugin.domain.repository.TaskRepository

/**
 * Use case for getting task response
 */
class GetResponseUseCase(
    private val repository: TaskRepository
) {
    suspend operator fun invoke(requestId: String): TaskInfo? {
        return repository.getTaskResponse(requestId)
    }
}

